loader.c

```diff
@@ -309,7 +308,16 @@ dloader_p api_load(const char *filename)
         case R_ARM_RELATIVE:
         {
             ElfW(Addr) *addr = (ElfW(Addr) *)(load_bias + reloc->r_offset);
-            *addr += load_bias;
+            if ((intptr_t) addr < ro_end && (intptr_t) addr >= ro_start)
+            {
+                mprotect((void*) round_down((intptr_t) addr, pagesize),
+                         pagesize, PROT_WRITE);
+                *addr += load_bias;
```

Comment:
The code is a bit tricky. Would it be working with Aarch64 as well?

Reply:
I've tried a build pass version with Aarch64.
Its also shows same things in .rela.dyn
```
Relocation section '.rela.dyn' at offset 0x328 contains 12 entries:
  Offset          Info           Type           Sym. Value    Sym. Name + Addend
0000000004d0  000000000403 R_AARCH64_RELATIV                    11048
0000000004d8  000000000403 R_AARCH64_RELATIV                    11050
0000000004e0  000000000403 R_AARCH64_RELATIV                    11040
```

Reply:
I need to do a full porting to Aarch64 to confirm this problem.